### PR TITLE
[9.0][MIG] medical medication

### PR DESCRIPTION
**Pull request by lasley**
This is dependent on:

- [x] #74 
- [x] #72

### CODE REVIEW

**Comment by woodbrettm** [comment]
This is good to go for review once medical_patient_disease is merged.

##### 1 Possible Issue
- In [medical_patient_medication_demo.xml](https://github.com/OCA/vertical-medical/pull/75/commits/684d4fcfdb227809435d8689ef7314609e6ad957#diff-a06e322829631336ff08d48a1cede16a), I had to declare medicament_id (required field) both for the medication_template and the medical_patient_medication models or else I'd get a null column integrity error. Not sure if this is by design, or a legitimate issue. Both medical_medication_template_template_1 and medical_patient_medication_medication_1 have a medicament_id field defined for example.

**Comment by lasley** [comment]
@bmw95 - please rebase now that disease is merged

File: medical_medication/tests/test_medical_medication_template.py
Comment (lasley) [inline medical_medication/tests/test_medical_medication_template.py:32]: No need to remove, but these strings are superfluous - the default fail says basically the same thing. If you name your methods well and have a descriptive first line of the docblock, you're good with just the default messages IMO.

**Review by lasley — COMMENTED** [review]
LGTM thanks @bmw95 - one note, but don't bother changing. It'll just save you a buttload of time in the future

**Comment by lasley** [comment]
@t3ddftw please review

File: medical_medication/tests/test_medical_medication_template.py
Comment (woodbrettm) [reply medical_medication/tests/test_medical_medication_template.py:32]: Makes sense. I'd been doing that just to keep the test structure consistent with those done in medical, but will change for future upgrades I work on :)

**Comment by woodbrettm** [comment]
@lasley I noticed a lot of the original copyrights were removed. Do you want me to add those back in?

File: medical_medication/demo/medical_patient_demo.xml
Comment (woodbrettm) [inline medical_medication/demo/medical_patient_demo.xml]: Should be 6, not 5. Don't forget to fix in medical_medication_demo and in tests if needed.

**Comment by lasley** [comment]
> I noticed a lot of the original copyrights were removed

@bmw95 - Yes please. My bad, I thought we were past the modules that happened on. I think this or prescription is the highest up on the dependency chain with this issue.

**Review by tedsalmon — APPROVED** [review]
LGTM! :)

**Comment by woodbrettm** [comment]
Alright good to go!

**Review by lasley — COMMENTED** [review]
Thanks @bmw95 - please squash for merge

**Comment by lasley** [comment]
Oops that was a stupid request, I forgot we're the same repo and I can squash myself. Thanks @bmw95 - will merge after CI green

**Pull request merged**
